src/visitors/rules/custom/regRule.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable no-underscore-dangle */
+
+class RegRule {
+  static get rule() {
+    return 'reg-rule';
+  }
+
+  constructor(config) {
+    this.ruleSets = {};
+
+    for (const ruleName of Object.keys(config.rules || {})) {
+      const [treeLevel, field] = config.rules[ruleName].on.split('.');
```

Comment:
add basic check here in case `on` is undefined. Crash with some meaningful error